### ISSUE TITLE
New rule: no-unbound-method-callbacks

### ISSUE THREAD

**Issue by danvk** [palantir/tslint-react#63]
This is a mistake:

```ts
class MyComponent extends React.Component<{}, {}> {
  render() {
    return <button onClick={this.handleClick}>Click me!</button>;
    //                      ~~~~~~~~~~~~~~~~  passing unbound method
  }

  handleClick(e: any) {
    alert('You clicked!');
  }
}
```

There are many possible solutions, a few of which include:

- `onClick={this.handleClick.bind(this)}`
- `onClick={e => this.handleClick(e)}`
- Adding `this.handleClick = this.handleClick.bind(this);` in the constructor

A lint rule would need to at least check for the last one to avoid false positives.

**Comment by adidahiya** [comment]
this could potentially be a more general rule: https://github.com/palantir/tslint/issues/259

**Comment by nchen63** [comment]
this will be part of the next tslint release https://github.com/palantir/tslint/pull/2089

**Comment by adidahiya** [comment]
closing as this is available in tslint core rules
